python_modules/dagster/dagster/_core/types/dagster_type.py

```diff
@@ -190,15 +192,18 @@ def from_builtin_enum(builtin_enum) -> "DagsterType":
     def metadata_entries(self) -> t.List[MetadataEntry]:
         return self._metadata_entries  # type: ignore
 
+    @public  # type: ignore
     @property
     def required_resource_keys(self) -> TypingAbstractSet[str]:
```

Comment:
naming here is rough

Reply:
yeah I agree, though alas I think the ship has sailed on this one